lib/active_fedora/schema_indexing_strategy.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveFedora
+  ##
+  # An application strategy to also apply solr indexes for each property.
+  # @note If how a field is indexed changes based on property, this would be a
+  #   good place to define that.
+  class SchemaIndexingStrategy
+    # @param [Array<Symbol>] index_types Ways to index all properties.
+    def initialize(index_types=[])
```

Comment:
This is a good point, but the #index returning a lambda interface feels weird. Thinking on alternatives.